packages/airnode-node/src/evm/authorization/authorization-fetching.ts

```diff
@@ -114,15 +142,27 @@ export async function fetch(
   }
 
   // If there are no authorizer contracts then endpoint is public
-  if (isEmpty(fetchOptions.authorizers)) {
+  if (isEmpty(fetchOptions.authorizers.requesterEndpointAuthorizers)) {
     const authorizationByRequestIds = apiCalls.map((pendingApiCall) => ({
       [pendingApiCall.id]: true,
     }));
     return [[], Object.assign({}, ...authorizationByRequestIds) as AuthorizationByRequestId];
   }
 
+  // Skip fetching authorization statuses if found in config for a specific authorization type
+  // and requester address
+  const { configAuthorizationsByRequestId, configAuthorizationRequestIds } = checkConfigAuthorizations(
```

Comment:
Do you actually need both `configAuthorizationsByRequestId` and `configAuthorizationRequestIds` as return values? Isn't `configAuthorizationRequestIds` just `Object.keys(configAuthorizationsByRequestId)`?

Reply:
> Isn't configAuthorizationRequestIds just Object.keys(configAuthorizationsByRequestId)

Not exactly as `configAuthorizationsByRequestId` is an array of objects so I thought this was a neater solution than mapping through `configAuthorizationsByRequestId` and getting each object's key, but this can also be done inline, fixed :smile:

Reply:
Oh, I see now. The reason why the `successfulResults` (at the end of the `fetch` function) is an array is that we chunk the authorization requests. You actually don't need to create an array (`AuthorizationByRequestId[]`) but just one object (`AuthorizationByRequestId`) in the `checkConfigAuthorizations` function. It will simplify `checkConfigAuthorizations` itself and the subsequent filtering as well.

Reply:
Very elegant, fixed!